### ISSUE TITLE
Does not build under Mac OS X El Capitan with homebrew 1.7.1

### ISSUE THREAD

**Issue by cpansprout** [rkitover/net-ssh2#44]
This is the error I am getting from `perl Makefile.PL`:
```
Looking for libraries...
Unable to find a working version of library ssh2 in the following directories:
  /usr/lib
  /usr/local/lib
  /usr/local
  /
  /usr
  /opt
  /opt/local
  /usr/local/libssh2
  /opt/libssh2
  /usr/local/ssl
  /Users/sprout
  /Users/sprout/libssh2
  /usr/local/Cellar/openssl/1.0.2o_2
```

I had a stab at trying to change Makefile.PL to recognise the libssh2 I have installed, but didn’t get very far. I don’t really know what I’m doing.

Eventually I tried this:
```
$ perl Makefile.PL lib=/usr/local/Cellar/libssh2/1.8.0/lib inc=/usr/local/Cellar/libssh2/1.8.0/include/libssh2:/usr/local/Cellar/openssl/1.0.2o_2/include
```
and Makefile.PL ran to completion, but `make` failed, with a compiler error:
```
SSH2.xs:52:10: fatal error: 'openssl/crypto.h' file not found
#include <openssl/crypto.h>
         ^
1 error generated.
```
Here is a list of the files in the relevant Cellar subdirectories (skipping the man pages):
```
$ find /usr/local/Cellar/libssh2/
/usr/local/Cellar/libssh2/
/usr/local/Cellar/libssh2//1.8.0
/usr/local/Cellar/libssh2//1.8.0/.brew
/usr/local/Cellar/libssh2//1.8.0/.brew/libssh2.rb
/usr/local/Cellar/libssh2//1.8.0/ChangeLog
/usr/local/Cellar/libssh2//1.8.0/COPYING
/usr/local/Cellar/libssh2//1.8.0/include
/usr/local/Cellar/libssh2//1.8.0/include/libssh2.h
/usr/local/Cellar/libssh2//1.8.0/include/libssh2_publickey.h
/usr/local/Cellar/libssh2//1.8.0/include/libssh2_sftp.h
/usr/local/Cellar/libssh2//1.8.0/INSTALL_RECEIPT.json
/usr/local/Cellar/libssh2//1.8.0/lib
/usr/local/Cellar/libssh2//1.8.0/lib/libssh2.1.dylib
/usr/local/Cellar/libssh2//1.8.0/lib/libssh2.a
/usr/local/Cellar/libssh2//1.8.0/lib/libssh2.dylib
/usr/local/Cellar/libssh2//1.8.0/lib/pkgconfig
/usr/local/Cellar/libssh2//1.8.0/lib/pkgconfig/libssh2.pc
/usr/local/Cellar/libssh2//1.8.0/NEWS
/usr/local/Cellar/libssh2//1.8.0/README
$ find /usr/local/Cellar/openssl/
/usr/local/Cellar/openssl/
/usr/local/Cellar/openssl//.DS_Store
/usr/local/Cellar/openssl//1.0.2o_2
/usr/local/Cellar/openssl//1.0.2o_2/.bottle
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/certs
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/misc
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/misc/c_hash
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/misc/c_info
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/misc/c_issuer
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/misc/c_name
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/misc/CA.pl
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/misc/CA.sh
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/misc/tsget
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/openssl.cnf
/usr/local/Cellar/openssl//1.0.2o_2/.bottle/etc/openssl/private
/usr/local/Cellar/openssl//1.0.2o_2/.brew
/usr/local/Cellar/openssl//1.0.2o_2/.brew/openssl.rb
/usr/local/Cellar/openssl//1.0.2o_2/bin
/usr/local/Cellar/openssl//1.0.2o_2/bin/c_rehash
/usr/local/Cellar/openssl//1.0.2o_2/bin/openssl
/usr/local/Cellar/openssl//1.0.2o_2/CHANGES
/usr/local/Cellar/openssl//1.0.2o_2/include
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/aes.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/asn1.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/asn1_mac.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/asn1t.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/bio.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/blowfish.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/bn.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/buffer.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/camellia.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/cast.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/cmac.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/cms.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/comp.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/conf.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/conf_api.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/crypto.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/des.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/des_old.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/dh.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/dsa.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/dso.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/dtls1.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/e_os2.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ebcdic.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ec.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ecdh.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ecdsa.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/engine.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/err.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/evp.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/hmac.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/idea.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/krb5_asn.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/kssl.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/lhash.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/md4.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/md5.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/mdc2.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/modes.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/obj_mac.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/objects.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ocsp.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/opensslconf.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/opensslv.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ossl_typ.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/pem.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/pem2.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/pkcs12.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/pkcs7.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/pqueue.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/rand.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/rc2.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/rc4.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ripemd.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/rsa.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/safestack.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/seed.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/sha.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/srp.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/srtp.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ssl.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ssl2.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ssl23.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ssl3.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/stack.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/symhacks.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/tls1.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ts.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/txt_db.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ui.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/ui_compat.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/whrlpool.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/x509.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/x509_vfy.h
/usr/local/Cellar/openssl//1.0.2o_2/include/openssl/x509v3.h
/usr/local/Cellar/openssl//1.0.2o_2/INSTALL_RECEIPT.json
/usr/local/Cellar/openssl//1.0.2o_2/lib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/lib4758cca.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libaep.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libatalla.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libcapi.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libchil.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libcswift.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libgmp.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libgost.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libnuron.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libpadlock.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libsureware.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/engines/libubsec.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/libcrypto.1.0.0.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/libcrypto.a
/usr/local/Cellar/openssl//1.0.2o_2/lib/libcrypto.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/libssl.1.0.0.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/libssl.a
/usr/local/Cellar/openssl//1.0.2o_2/lib/libssl.dylib
/usr/local/Cellar/openssl//1.0.2o_2/lib/pkgconfig
/usr/local/Cellar/openssl//1.0.2o_2/lib/pkgconfig/libcrypto.pc
/usr/local/Cellar/openssl//1.0.2o_2/lib/pkgconfig/libssl.pc
/usr/local/Cellar/openssl//1.0.2o_2/lib/pkgconfig/openssl.pc
/usr/local/Cellar/openssl//1.0.2o_2/LICENSE
/usr/local/Cellar/openssl//1.0.2o_2/NEWS
/usr/local/Cellar/openssl//1.0.2o_2/README
```

**Comment by rkitover** [comment]
I'm not sure the inc param parses multiple dirs.

One thing you can try, try running:

`brew link --force openssl`

before running the Makefile.PL

and then running:

`brew unlink openssl`

once everything is installed (or not.)

**Comment by cdalvaro** [comment]
Probably, this issue must have been solved in #50

**Comment by rkitover** [comment]
This was still 1.0, so does not seem directly related. Hopefully whatever it was is no longer an issue.
